src/pages/MyCakesList.js

```diff
@@ -0,0 +1,52 @@
+import "./MyCakesList.css";
+import { Link } from "react-router-dom";
+import { RiFilePaper2Line } from "react-icons/ri";
+import ChCakeImg from "../images/chocolate.png";
+
+export default function MyCakesList() {
+  const recipes = JSON.parse(localStorage.getItem("recipesArray"));
+
+  function renderCakes() {
+    return recipes.map((cake, index) => {
+      const id = index + 1;
+      return (
+        <div className="ItemCard box">
+          <li className="ItemCard" key={cake.id}>
+            <img
+              src={ChCakeImg}
+              alt="description"
+              className="ItemCard__image"
+            />
+
+            <h2 className="ItemCard__name">{cake.cakeName}</h2>
+            <div className="ItemCard--btnDiv">
+              <button className="ItemCard__button delete">X</button>
+
+              <Link to={`/cakes/${id}`}>
```

Comment:
you are using two different id's here. it's confusing but also I not see at this point why you not use the same?